Sources/App/Controllers/FormatAPIController.swift

```diff
@@ -0,0 +1,222 @@
+import Foundation
+import Vapor
+
+class FormatAPIController: RouteCollection {
+    
+    private let calendars = [
+        "gregorian": Calendar(identifier: .gregorian),
+        "buddhist": Calendar(identifier: .buddhist),
+        "chinese": Calendar(identifier: .chinese),
+        "coptic": Calendar(identifier: .coptic),
+        "ethiopicAmeteMihret": Calendar(identifier: .ethiopicAmeteMihret),
+        "ethiopicAmeteAlem": Calendar(identifier: .ethiopicAmeteAlem),
+        "hebrew": Calendar(identifier: .hebrew),
+        "iso8601": Calendar(identifier: .iso8601),
+        "indian": Calendar(identifier: .indian),
+        "islamic": Calendar(identifier: .islamic),
+        "islamicCivil": Calendar(identifier: .islamicCivil),
+        "japanese": Calendar(identifier: .japanese),
+        "persian": Calendar(identifier: .persian),
+        "republicOfChina": Calendar(identifier: .republicOfChina),
+        "islamicTabular": Calendar(identifier: .islamicTabular),
+        "islamicUmmAlQura": Calendar(identifier: .islamicUmmAlQura)
+    ]
+    
+    func boot(routes: RoutesBuilder) throws {
+        routes.get("info.json", use: infoJSON)
+        routes.post("format.json", use: formatJSON)
+    }
+    
+    func infoJSON(_ req: Request) async throws -> InfoResponse {
+        return InfoResponse(calendars: self.calendars.keys.sorted(by: <),
+                            locales: Locale.availableIdentifiers.sorted(by: <),
+                            timeZones: TimeZone.knownTimeZoneIdentifiers.sorted(by: <),
+                            timeZoneDataVersion: TimeZone.timeZoneDataVersion)
+    }
+    
+    func formatJSON(_ req: Request) async throws -> [FormatResponse] {
+        let requests = try req.content.decode([FormatRequest].self)
+        return try requests.map(processRequest)
+    }
+    
+    private func processRequest(_ request: FormatRequest) throws -> FormatResponse {
+        let locale: Locale
+        let calendar: Calendar
+        let calendarID: String
+        let timeZone: TimeZone
+        
+        if let localeID = request.locale {
+            let loweredLocaledID = localeID.lowercased()
+            guard let actualLocaleID = Locale.availableIdentifiers.first(where: { $0.lowercased() == loweredLocaledID }) else {
+                throw Abort(.badRequest, reason: "Unknown locale: '\(localeID)'")
+            }
+            locale = Locale(identifier: actualLocaleID)
+        } else {
+            locale = Locale(identifier: "en_US_POSIX")
+        }
```

Comment:
perhaps cleaner to extract this and the time zone parsing / defaults into a separate method?

```swift
private func resolveLocale(_ request: FormatRequest) throws -> Locale {
    if let localeID = request.locale {
            let loweredLocaledID = localeID.lowercased()
            guard let actualLocaleID = Locale.availableIdentifiers.first(where: { $0.lowercased() == loweredLocaledID }) else {
                throw Abort(.badRequest, reason: "Unknown locale: '\(localeID)'")
            }
            locale = Locale(identifier: actualLocaleID)
        } else {
            locale = Locale(identifier: "en_US_POSIX")
        }
}
```

Then the callsite becomes pretty clean:

```swift
let locale = try resolveLocale(request)
let timeZone = try resolveTimeZone(request) // e.g.
```